overlord/ifacestate/ifacemgr.go

```diff
@@ -152,8 +152,12 @@ var assessAppArmorPrompting = func(m *InterfaceManager) bool {
 // InterfacesRequestsManager returns the interfaces requests manager associated
 // with the receiver. This method may only be called after StartUp has been
 // called, and will return nil if AppArmor prompting is not running.
-func (m *InterfaceManager) InterfacesRequestsManager() *apparmorprompting.InterfacesRequestsManager {
-	return m.interfacesRequestsManager
+func (m *InterfaceManager) InterfacesRequestsManager() apparmorprompting.Manager {
+	irm := m.interfacesRequestsManager
+	if irm == nil {
+		return nil
+	}
+	return irm
```

Comment:
Was this that tricky issue where the interface type variable (one level up in this cas) doesn't equal `nil` because it has a dynamic type although the value is nil?

Reply:
yes